lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -868,6 +878,21 @@ def exec_command(self, command):
             stderr=subprocess.DEVNULL,
         )
 
+    def detect_init_system(self):
+        """Detect whether services are running with Docker or systemd."""
+
+        docker_running = "magmad" in self.exec_command_output("docker ps --filter 'name=magmad' --format '{{.Names}}'")
+        systemd_running = self.exec_command("systemctl is-active magma@magmad") == 0
+
+        if docker_running and systemd_running:
+            raise AssertionError("Magmad is running with both Docker and systemd.")
```

Comment:
nitpick: I think `AssertionError is really just meant to be raised by `assert`. When manually raising an exception there is usually something more specific that fits better. That said, I'm not sure what would be a good one here ;)

Reply:
I think people often use `ValueError` for generic precondition errors like here even if it is meant for something more specific (strictly-speaking). Another option would be a generic `RuntimeError`.

Reply:
Not sure if this qualifies for `OSError`, otherwise a more general one would be `RuntimeError`

Reply:
There's also an `EnvironmentError` that's an alias of `OSError` that might fit. On the other hand, `OSError` refers to I/O issues in its docstring and that doesn't really fit...

Reply:
Thanks for pointing this out and for the suggestions! To me `EnvironmentError` makes the most intuitive sense but it seems to have been deprecated, so I think I'll just go for `RuntimeError`.